src/renderer/themedSplash.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * Vencord Desktop, a desktop app aiming to give you a snappier Discord Experience
+ * Copyright (c) 2023 Vendicated and Vencord contributors
+ */
+
+import { Settings } from "./settings";
+
+window.addEventListener("load", () => {
```

Comment:
Technically this shouldn't be necessary, but i would prefer if you checked document.readyState first to make sure this callback always runs even if the document is somehow already loaded

Much similar to here https://github.com/Vencord/Vesktop/blob/main/src/preload/index.ts#L29-L35

Reply:
Also, as mentioned on Discord, with this approach the variables won't be properly updated if you change theme in any way. One solution would be to use the beforeunload event to save them another time right before closing the app. Though I'm not sure if this is actually called if you Alt+F4 (or equivalent). If no, there are likely similar electron apis that can achieve this